core/repository/consensus/merkle_transaction_repository.hpp

```diff
@@ -1,16 +1,73 @@
+/*
+Copyright Soramitsu Co., Ltd. 2016 All Rights Reserved.
+http://soramitsu.co.jp
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
 #ifndef CORE_REPOSITORY_MERKLETRANSACTIONREPOSITORY_HPP_
 #define CORE_REPOSITORY_MERKLETRANSACTIONREPOSITORY_HPP_
 
 #include <string>
 #include <memory>
+#include <unordered_map>
 #include "../../model/transactions/abstract_transaction.hpp"
-
 #include "../../consensus/consensus_event.hpp"
 
 namespace merkle_transaction_repository {
-//bool commit(std::string const hash, std::unique_ptr<consensus_event::ConsensusEvent> const tx);
-bool commit(std::string, const std::unique_ptr<consensus_event::ConsensusEvent> &event);
-std::unique_ptr<abstract_transaction::AbstractTransaction> findLeaf(std::string const hash);
+
+struct MerkleNode {
+    std::string hash;
+    std::string parent;
+    std::tuple<std::string, std::string> children;
+
+    MerkleNode(std::unordered_map<std::string, std::string> translateJSON) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Struct 'MerkleNode' has a constructor with 1 argument that is not explicit.](https://www.codacy.com/app/soramitsu/iroha/file/4318160647/issues/source?bid=3712501&fileBranchId=3857189#l34)